fastplotlib/graphics/line.py

```diff
@@ -83,3 +83,35 @@ def __init__(
 
         if z_position is not None:
             self.world_object.position.z = z_position
+
+    def _set_feature(self, feature: str, new_data: Any, indices: Any = None):
+        if not hasattr(self, "_previous_data"):
+            self._previous_data = {}
+        elif hasattr(self, "_previous_data"):
+            self._reset_feature(feature)
+        # if feature in self._feature_events:
+        feature_instance = getattr(self, feature)
+        if indices is not None:
+            previous = feature_instance[indices].copy()
+            feature_instance[indices] = new_data
+        else:
+            previous = feature_instance[:].copy()
+            feature_instance[:] = new_data
+        if feature in self._previous_data.keys():
+            self._previous_data[feature].previous_data = previous
+            self._previous_data[feature].previous_indices = indices
+        else:
+            self._previous_data[feature] = PreviouslyModifiedData(previous_data=previous, previous_indices=indices)
+        # else:
+        #     raise ValueError("name arg is not a valid feature")
+
+    def _reset_feature(self, feature: str):
+        if feature not in self._previous_data.keys():
+            raise ValueError("no previous data registered for this feature")
+        else:
+            feature_instance = getattr(self, feature)
+            if self._previous_data[feature].previous_indices is not None:
+                feature_instance[self._previous_data[feature].previous_indices] = self._previous_data[
+                    feature].previous_data
+            else:
+                feature_instance[:] = self._previous_data[feature].previous_data
```

Comment:
similar to previous comment, use this which works for both indexable and non-indexable:

```python
feature_instance._set(...)
```